pkg/controllers/vdb/passwordsecret_reconciler.go

```diff
@@ -86,26 +90,48 @@ func (a *PasswordSecretReconciler) statusMatchesSpec() bool {
 	return a.Vdb.Spec.PasswordSecret == *a.Vdb.Status.PasswordSecret
 }
 
-// updatePasswordSecret will update the password secret in the database
+// updatePasswordSecret will update the password secret in the database. It starts with the main cluster,
+// then cycles through the sandboxes.
 func (a *PasswordSecretReconciler) updatePasswordSecret(ctx context.Context) (ctrl.Result, error) {
-	// The password to be updated to
+	// Get new password; don't look in cache because it has old secret
 	newPasswd, err := vk8s.GetCustomSuperuserPassword(ctx, a.VRec.Client, a.Log, a.VRec, a.Vdb,
-		a.Vdb.Spec.PasswordSecret, names.SuperuserPasswordKey)
+		a.Vdb.Spec.PasswordSecret, names.SuperuserPasswordKey, a.PasswordManager, true)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
+	res, err := a.updatePasswordSecretInSandbox(ctx, a.PFacts, newPasswd, "")
+	if verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+	for _, sb := range a.Vdb.Spec.Sandboxes {
```

Comment:
Could we let sandbox controller handle sandbox password change? When password secret changes, we first change the password in the main cluster and then for each sandbox, we wake up the sandbox controller which will change the password in each sandbox. That way, sandboxes password change will be done concurrently, and one failure will not affect other sandboxes.
The same logic as sandbox upgrade.